src/components/com_tjcertificate/administrator/controllers/trainingrecord.json.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * @package     TJCertificate
+ * @subpackage  com_tjcertificate
+ *
+ * @author      Techjoomla <extensions@techjoomla.com>
+ * @copyright   Copyright (C) 2009 - 2020 Techjoomla. All rights reserved.
+ * @license     http://www.gnu.org/licenses/gpl-2.0.html GNU/GPL
+ */
+
+// No direct access to this file
+defined('_JEXEC') or die('Restricted access');
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\Language\Text;
+use Joomla\Registry\Registry;
+use Joomla\Utilities\ArrayHelper;
+use Joomla\CMS\MVC\Controller\FormController;
+use Joomla\CMS\Filesystem\Folder;
+use Joomla\CMS\Filesystem\File;
+use Joomla\CMS\Component\ComponentHelper;
+use Joomla\CMS\Session\Session;
+use Joomla\CMS\Router\Route;
+use Joomla\CMS\Response\JsonResponse;
+use Joomla\CMS\Table\Table;
+
+JLoader::import("/techjoomla/media/storage/local", JPATH_LIBRARIES);
+
+/**
+ * The Tj Certificate Training Record controller
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class TjCertificateControllerTrainingRecord extends FormController
+{
+	/**
+	 * Function to delete the record attachment
+	 *
+	 * @return  void
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public function deleteAttachment()
+	{
+		$app = Factory::getApplication();
+
+		if (!Session::checkToken())
+		{
+			$app->enqueueMessage(Text::_('JINVALID_TOKEN'), 'error');
+			echo new JsonResponse(null, null, true);
+			$app->close();
+		}
+
+		// Get the current user id
+		$user = Factory::getuser();
+
+		if (!$user->id)
+		{
+			return false;
+		}
+
+		$filePath = 'media/com_tjcertificate/external';
+		$clientId = $app->input->get('certificateId', '', 'INT');
+		$mediaId  = $app->input->get('mediaId', '', 'INT');
```

Comment:
Default values should be 0 for INT. Change for both $clientId & $mediaId

Reply:
done